### PR TITLE
chore(inputs.procstat): Convert bool tagging settings to a list

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

based-on #14329

This PR converts the boolean flags for which information to use as a tag to a string list which is more extensible,

### CODE REVIEW

File: migrations/inputs_procstat/migration_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf/config"
+	_ "github.com/influxdata/telegraf/migrations/inputs_procstat" // register migration
+	_ "github.com/influxdata/telegraf/plugins/inputs/procstat"    // register plugin
+)
+
+func TestNoMigration(t *testing.T) {
+	defaultCfg := []byte(`
```
Comment (Hipska) [inline migrations/inputs_procstat/migration_test.go:16]: Replace by using `(*Procstat).SampleConfig()`?

**Review by Hipska — APPROVED** [review]

**Comment by powersj** [comment]
@Hipska,

@srebhan and I use the `ready for final review` tag to mean the other has given the +1 to a PR. We then assign it to the other, so we know who owns the next step, and to ensure we know who needs to press merge.

**Comment by Hipska** [comment]
I have seen you guys started to use this tag differently compared to originally discussed/agreed. (Like adding it to your own PRs without anyone giving it a first review) I don’t think this new way to use the tag has been informed to the community maintainers. Would be nice to inform us, so we can use it correctly.

**Comment by powersj** [comment]
> I have seen you guys started to use this tag differently compared to originally discussed/agreed

This is not a recent change, this is how Sven and I have used this since the layoff.

**Comment by powersj** [comment]
@srebhan please rebase tomorrow and we can finish these up! Thank you in advanced!

**Review by powersj — APPROVED** [review]
Thank you for this and all the clean up to procstat

**Pull request merged**